setup.py

```diff
@@ -71,7 +71,12 @@
         "blobfile~=3.0.0",
     ],
     extras_require={
-        "arrow": ["pyarrow>=13.0.0", "pandas~=2.0.0"],
+        "arrow": [
+            "pyarrow>=13.0.0",
+            "joblib~=1.4.2",
```

Comment:
let's if we could go without it (maybe ThreadingPool from multiprocessing will do the jobs)